heroic-core/src/main/java/com/spotify/heroic/consumer/schemas/Spotify100.java

```diff
@@ -179,6 +185,76 @@ public Consumer(Clock clock, IngestionGroup ingestion, ConsumerReporter reporter
             AsyncFuture<Ingestion> ingestionFuture =
                 ingestion.write(new Request(series, MetricCollection.points(points)));
 
+            // Return Void future, to not leak unnecessary information from the backend but just
+            // allow monitoring of when the consumption is done.
+            return ingestionFuture.directTransform(future -> null);
+        }
+
+        private AsyncFuture<Void> handleVersion2(final JsonNode tree)
+            throws ConsumerSchemaValidationException {
+            final com.spotify.heroic.consumer.schemas.spotify100.v2.JsonMetric metric;
+
+            try {
+                metric = new TreeTraversingParser(tree, mapper)
+                    .readValueAs(
+                        com.spotify.heroic.consumer.schemas.spotify100.v2.JsonMetric.class);
+            } catch (IOException e) {
+                throw new ConsumerSchemaValidationException("Invalid metric", e);
+            }
+
+            if (metric.getValue() == null) {
+                throw new ConsumerSchemaValidationException(
+                    "Metric must have a value but this metric has a null value: " + metric);
+            }
+
+            if (metric.getTime() == null) {
+                throw new ConsumerSchemaValidationException("time: field must be defined: " + tree);
+            }
+
+            if (metric.getTime() <= 0) {
+                throw new ConsumerSchemaValidationException(
+                    "time: field must be a positive number: " + tree);
+            }
+
+            if (metric.getKey() == null) {
```

Comment:
70% of `handleVersion2()` is copy pasta from `handleVersion1()`. Please either explain why this is, or refactor the common 70%.

Reply:
Agreed, I thought about it but decided not to change anything. it is too much work for little return.
We need to eventually take on the task to streamline Metric definition may be refactoring can be done with that task.

Reply:
I just don't agree, to be honest.

Reply:
Sorry but I just don't agree. Copy pasted code needs a very strong argument for keeping it.

Reply:
Alright, I hope you don't feel like the entire PR was a copy and past :)